bottle_app.py

```diff
@@ -629,14 +629,59 @@ def api_get_movie_trackpoints():
         labels         = sorted( ( tp['label'] for tp in trackpoint_dicts) )
         frame_dicts    = defaultdict(dict)
 
+        ruler0_x, ruler20_x = -1, -1
+        ruler0_y, ruler20_y = -1, -1
+        ruler0_frame_number = -1
+        sensor_width_mm = 36  # Sensor width 36 millimeters (this is a common value,
+        # but should be adjusted according to actual circumstances)
+        actual_distance_mm = 20
+        mm_per_inch = 25.4  # common value
+
         for tp in trackpoint_dicts:
-            frame_dicts[tp['frame_number']][tp['label']+' x'] = tp['x']
-            frame_dicts[tp['frame_number']][tp['label']+' y'] = tp['y']
+            frame_dicts[tp['frame_number']][tp['label'] + ' x'] = tp['x']
+            frame_dicts[tp['frame_number']][tp['label'] + ' y'] = tp['y']
```

Comment:
Here you put a lot of business logic for converting from screen positions to actual coordinates in the file `bottle_app.py`. I do not think that this code should be in this file. Please move the business logic function to `tracker.py`. The function should take three (x,y) points: the two points of the ruler and the point that we are moving to mm coordinates. It should return the (x,y) of the point in mm using the 0mm point of the ruler as the origin of the coordinate system.

When you submit that function, please also provide a test for that function.